python/cudf/cudf/tests/test_udf_masked_ops.py

```diff
@@ -673,3 +673,16 @@ def f(x):
     data.apply(f)
 
     assert precompiled.currsize == 1
+
+
+@pytest.mark.parametrize(
+    "data", [[1.0, 0.0, 1.5], [1, 1, 2], [True, False, True]]
+)
+@pytest.mark.parametrize("operator", [float, int, bool])
+def test_masked_udf_casting(operator, data):
+    data = cudf.Series(data)
+
+    def func(x):
+        return operator(x)
```

Comment:
Can we just pass `operator` directly as the function, or is this level of indirection (wrapped in a function) necessary?

Reply:
We need it to be a function so that it can have a `__code__` attribute, this is used to generate the cache key later down the line. We could maybe add some special casing for this but it would be basically enabling this:

```
sr = cudf.Series([1,2,3])
sr.apply(float)
```
which feels like just a way more complicated way of calling `astype`.

Reply:
Ah! I think I've asked about this kind of thing before, relating to the reliance on `__code__` attributes in the UDF machinery. Not all functions have a `__code__` attribute, and I think this should be permissible/supported behavior. It's fine if we don't do it in this PR and file an issue instead, but it should _work_ one way or another, even if it means that we simply skip using the cache or have to wrap it.